modules/index/src/main/java/org/apache/ignite/internal/idx/IndexManagerImpl.java

```diff
@@ -0,0 +1,409 @@
+import org.apache.ignite.internal.util.IgniteSpinBusyLock;
+import org.apache.ignite.lang.IgniteException;
+import org.apache.ignite.lang.IgniteLogger;
+import org.apache.ignite.lang.IgniteUuid;
+import org.apache.ignite.lang.IndexAlreadyExistsException;
+import org.apache.ignite.lang.IndexNotFoundException;
+import org.apache.ignite.lang.NodeStoppingException;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Internal index manager facade provides low-level methods for indexes operations.
+ */
+public class IndexManagerImpl extends AbstractProducer<IndexEvent, IndexEventParameters> implements IndexManager, IgniteComponent {
+    private static final IgniteLogger LOG = IgniteLogger.forClass(IndexManagerImpl.class);
+
+    private final TablesConfiguration tablesCfg;
+
+    private final TableManager tblMgr;
+
+    /** Indexes by canonical name. */
+    private final Map<String, InternalSortedIndex> idxsByName = new ConcurrentHashMap<>();
+
+    /** Busy lock to stop synchronously. */
+    private final IgniteSpinBusyLock busyLock = new IgniteSpinBusyLock();
+
+    /** Prevents double stopping the component. */
+    private final AtomicBoolean stopGuard = new AtomicBoolean();
+
+    /**
+     * Constructor.
+     *
+     * @param tablesCfg      Tables configuration.
+     */
+    public IndexManagerImpl(
+            TableManager tblMgr,
+            TablesConfiguration tablesCfg
+    ) {
+        this.tblMgr = tblMgr;
+        this.tablesCfg = tablesCfg;
+    }
+
+    @Override
+    public void start() {
+        tablesCfg.tables().any().indices().listenElements(
+                new ConfigurationNamedListListener<>() {
+                    @Override
+                    public @NotNull CompletableFuture<?> onCreate(@NotNull ConfigurationNotificationEvent<TableIndexView> ctx) {
+                        TableConfiguration tbl = ctx.config(TableConfiguration.class);
+                        String tblName = tbl.name().value();
+                        String idxName = ctx.newValue().name();
+
+                        if (!busyLock.enterBusy()) {
+                            fireEvent(IndexEvent.CREATE,
+                                    new IndexEventParameters(
+                                            idxName,
+                                            tblName),
+                                    new NodeStoppingException());
+
+                            return CompletableFuture.completedFuture(new NodeStoppingException());
+                        }
+                        try {
+                            onIndexCreate(ctx);
+                        } catch (Exception e) {
+                            fireEvent(
+                                    IndexEvent.CREATE,
+                                    new IndexEventParameters(
+                                            idxName,
+                                            tblName),
+                                    e
```

Comment:
strange formatting and if it`s ok bu other check style it`s no ok from human readable practice ..